src/ardrive.ts

```diff
@@ -422,11 +423,11 @@ export class ArDrive extends ArDriveAnonymous {
 
 		const originalFolderMetaData = await this.getPrivateFolder(folderId, driveKey);
 
-		if (destFolderDriveId !== originalFolderMetaData.driveId) {
+		if (!destFolderDriveId.isEqualTo(originalFolderMetaData.driveId)) {
 			throw new Error(errorMessage.cannotMoveToDifferentDrive);
 		}
 
-		if (originalFolderMetaData.parentFolderId === newParentFolderId) {
+		if (`${originalFolderMetaData.parentFolderId}` === `${newParentFolderId}`) {
```

Comment:
Same here, parentFolderId is nullable (?)
```suggestion
		if (originalFolderMetaData.parentFolderId.isEqualTo(newParentFolderId)) {
```